pennylane/devices/qutrit_mixed/apply_operation.py

```diff
@@ -0,0 +1,276 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions to apply operations to a qutrit mixed state."""
+# pylint: disable=unused-argument
+
+import functools
+from functools import singledispatch
+from string import ascii_letters as alphabet
+import pennylane as qml
+from pennylane import math
+from pennylane import numpy as np
+from pennylane.operation import Channel
+
+alphabet_array = np.array(list(alphabet))
+
+EINSUM_OP_WIRECOUNT_PERF_THRESHOLD = 3
+EINSUM_STATE_WIRECOUNT_PERF_THRESHOLD = 13  # TODO placeholder value, need to ask how to find these
```

Comment:
I'm not sure how these were found for qubits, these are placeholder values right now.

Reply:
I believe this was a manual hyperparameter optimization. We could do a similar benchmarking for `qutrit` to see what makes sense.

Reply:
Depending on how much we care about performance optimization, we could just go with one of the methods for all cases. Right now, how much do we care about "it works" versus "it makes performant choices"?  I would be ok with focusing on "it works" and ignoring the performance tuning for now.

In the original PR: https://github.com/PennyLaneAI/pennylane/pull/3637 , I attached a script I used with some performance numbers.  For the `EINSUM_STATE_WIRECOUNT_PERF_THRESHOLD`, the relevant code was:

```
num_wires_array = [11,12,13,14,15,16,17]

for ni in num_wires_array:
    print("number of wires: ", ni)
    state = np.zeros(2**ni, dtype=np.complex128)
    state[0] = 1.0
    state.shape = [2]*ni
    
    op = qml.Rot(1.2, -0.2, 0.5, wires=10)
    %timeit apply_operation_einsum(op, state)
    %timeit apply_operation_tensordot(op, state)
    
    print("")
```

The idea is to create a state of a certain size, and then apply an operation using einsum and apply it using tensordot.  At some point, the performance may switch from one to the other.  For density matrices, we will probably be looking at much smaller sizes before the switch over.

Reply:
Thanks for the details @albi3ro ! Right now we care more about the "it works" aspect; I asked @Gabriel-Bottrill to leave notes where there are potential optimizations which can be looked into later, so we will add this kind of benchmarking to the list.

Reply:
It may then be wise ok to remove tensordot entirely for now as the code is available for later and it will simplify the PR a decent amount. 

One question for that that I have though is:
If I do that however I am wondering if it's worth also not testing both `apply_operation_einsum` and `apply_operation` seperately using the pytest parameter `method` or if it's ok to leave it as is for future additions.

Reply:
@albi3ro @glassnotes  do you think I should remove `apply_operation_tensordot` for now?